[Emacs 30.1] Add configuration for global-completion-preview

jeffbowman opened the pull request:
Adds configuration for global-completion-preview in the crafted-defaults-config module. 

Closes #431

jeffbowman committed:
[Emacs 30.1] Add configuration for global-completion-preview

sthesing commented:
Just a quick question: what if a user uses `crafted-completion-config` and has `corfu` installed. Don't they interfere? Shouldn't we turn `global-completion-preview` off in the corfu config?
I won't be able to test until tomorrow. But maybe you already thought of that and tried it?

jeffbowman commented:
Ah.. good point, I should turn that off, I'll update the PR.

jeffbowman committed:
[Emacs 30.1] Turn off global-completion-preview-mode

Turn off global-completion-preview-mode in crafted-completion-config
with other built-in modes related to completion.

sthesing commented on modules/crafted-completion-config.el:
```diff
@@ -30,7 +30,11 @@
     (icomplete-mode -1)
-    (icomplete-vertical-mode -1)))
+    (icomplete-vertical-mode -1)
+    (when (version< "30" emacs-version)
+      ;; this mode is only available in Emacs version 30.1 and
+      ;; greater.
+      (global-completion-preview-mode -1))))
```
I think this should not go here in the `vertico`-config, but  in the code behind the `(when (require 'corfu nil :noerror)` guard.

jeffbowman committed:
[Emacs 30.1] Refactor turning off global-completion-preview-mode

Move turning off global completion preview mode to the configuration
for Corfu instead of Vertico.

sthesing reviewed (approved):
Thanks!
